cli/ml_inference.go

```diff
@@ -0,0 +1,119 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/urfave/cli/v2"
+	v1 "go.viam.com/api/app/data/v1"
+	mlinferencepb "go.viam.com/api/app/mlinference/v1"
+)
+
+const (
+	inferenceFlagFileOrgID      = "file-org-id"
+	inferenceFlagFileID         = "file-id"
+	inferenceFlagFileLocationID = "file-location-id"
+	inferenceFlagModelID        = "model-id"
+	inferenceFlagModelVersionID = "model-version"
+)
+
+type mlInferenceInferArgs struct {
+	OrgID          string
+	FileOrgID      string
+	FileID         string
+	FileLocationID string
+	ModelID        string
+	ModelVersion   string
+}
+
+// MLInferenceInferAction is the corresponding action for 'inference infer'.
+func MLInferenceInferAction(c *cli.Context, args mlInferenceInferArgs) error {
+	client, err := newViamClient(c)
+	if err != nil {
+		return err
+	}
+
+	_, err = client.mlRunInference(
+		args.OrgID, args.FileOrgID, args.FileID, args.FileLocationID,
+		args.ModelID, args.ModelVersion)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// mlRunInference runs inference on an image with the specified parameters.
+func (c *viamClient) mlRunInference(orgID, fileOrgID, fileID, fileLocation, modelID,
+	modelVersion string,
+) (*mlinferencepb.GetInferenceResponse, error) {
+	if err := c.ensureLoggedIn(); err != nil {
+		return nil, err
+	}
+
+	req := &mlinferencepb.GetInferenceRequest{
+		OrganizationId: orgID,
+		BinaryId: &v1.BinaryID{
+			FileId:         fileID,
+			OrganizationId: fileOrgID,
+			LocationId:     fileLocation,
+		},
+		RegistryItemId:      modelID,
+		RegistryItemVersion: modelVersion,
+	}
+
+	resp, err := c.mlInferenceClient.GetInference(context.Background(), req)
+	if err != nil {
+		return nil, errors.Wrapf(err, "received error from server")
+	}
+	c.printInferenceResponse(resp)
+	return resp, nil
+}
+
+// printInferenceResponse prints a neat representation of the GetInferenceResponse.
+func (c *viamClient) printInferenceResponse(resp *mlinferencepb.GetInferenceResponse) {
+	printf(c.c.App.Writer, "Inference Response:")
```

Comment:
You would know better than I would but is this the kind of format that you would want? Meaning, normally from a CLI I would expect JSON or something configurable that I can do programmatically. But I also know absolutely nothing about ML and this could be the way ML people want the response.

Reply:
Honestly, this format isn't very good for most usecases, but it's readable. I think the primary use case of this command will be to debug / easily check if the service is functioning.